rust/phoenix-sdk-core/src/sdk_client_core.rs

```diff
@@ -146,232 +156,233 @@ impl SDKClientCore {
     /// convert 3 Widgets to base lots you would call sdk.base_amount_to_base_lots(3_000_000_000). This would return
     /// the number of base lots that would be equivalent to 3 Widgets or 3 * 1e9 Widget atoms.
-    pub fn base_atoms_to_base_lots(&self, market_key: &Pubkey, base_atoms: u64) -> Result<u64> {
-        let market = self
+    pub fn base_atoms_to_base_lots_rounded_down(
+        &self,
+        market_key: &Pubkey,
+        base_atoms: u64,
+    ) -> Result<u64> {
+        let metadata = self
             .markets
             .get(market_key)
             .ok_or_else(|| anyhow!("Market not found! Please load in the market first."))?;
+        Ok(base_atoms / metadata.base_atoms_per_base_lot)
+    }
 
-        Ok(base_atoms / market.base_lot_size) // Lot size is the number of atoms in a lot
+    /// The same function as base_atoms_to_base_lots_rounded_down, but rounds up instead of down.
+    pub fn base_atoms_to_base_lots_rounded_up(
+        &self,
+        market_key: &Pubkey,
+        base_atoms: u64,
+    ) -> Result<u64> {
+        let metadata = self
+            .markets
+            .get(market_key)
+            .ok_or_else(|| anyhow!("Market not found! Please load in the market first."))?;
+        Ok(1 + base_atoms.saturating_sub(1) / metadata.base_atoms_per_base_lot)
     }
 
     /// RECOMMENDED:
     /// Converts base lots to base atoms. For example if the base currency was a Widget where there are
     /// 1_000 base atoms per base lot of Widget, you would call sdk.base_lots_to_base_atoms(300) to convert 300 base lots
     /// to 300_000 Widget atoms.
     pub fn base_lots_to_base_atoms(&self, market_key: &Pubkey, base_lots: u64) -> Result<u64> {
-        let market = self
+        let metadata = self
             .markets
             .get(market_key)
             .ok_or_else(|| anyhow!("Market not found! Please load in the market first."))?;
-        Ok(base_lots * market.base_lot_size) // Lot size is the number of atoms in a lot
+
+        Ok(base_lots * metadata.base_atoms_per_base_lot)
     }
 
     /// RECOMMENDED:
     /// Converts quote units to quote lots. For example if the quote currency was USDC you wanted to
     /// convert 3 USDC to quote lots you would call sdk.quote_unit_to_quote_lots(3.0). This would return
     /// the number of quote lots that would be equivalent to 3 USDC.
     pub fn quote_units_to_quote_lots(&self, market_key: &Pubkey, quote_units: f64) -> Result<u64> {
-        let market = self
+        let metadata = self
             .markets
             .get(market_key)
             .ok_or_else(|| anyhow!("Market not found! Please load in the market first."))?;
-        Ok((quote_units * market.quote_multiplier as f64 / market.quote_lot_size as f64) as u64)
+        Ok((quote_units
+            * (metadata.quote_atoms_per_quote_unit / metadata.quote_atoms_per_quote_lot) as f64)
```

Comment:
I think this should do everything as u64. `quote_atoms_per_quote_unit / quote_atoms_per_quote_lot` should always be an integer

Reply:
quote_units is a float though